internal/datanode/buffer.go

```diff
@@ -17,13 +17,220 @@
 package datanode
 
 import (
+	"container/heap"
 	"errors"
+	"fmt"
+	"github.com/milvus-io/milvus-proto/go-api/schemapb"
+	"github.com/milvus-io/milvus/internal/metrics"
 	"math"
+	"sync"
 
 	"github.com/milvus-io/milvus/internal/proto/datapb"
 	"github.com/milvus-io/milvus/internal/storage"
 )
 
+// DelBufferManager is in charge of managing insertBuf and delBuf from an overall prospect
+// not only controlling buffered data size based on every segment size, but also triggering
+// insert/delete flush when the memory usage of the whole manager reach a certain level.
+// but at the first stage, this struct is only used for delete buff
+type DelBufferManager struct {
+	delBufMap     sync.Map // map[segmentID]*DelDataBuf
+	delMemorySize int64
+	delBufHeap    *PriorityQueue
+}
+
```

Comment:
use priorityQueue to mange delDataBuf and the amount of memory used by every delDataBuf, in order to identify
correct segments set to flush for auto flushing